hyperspy/io_plugins/hdf5.py

```diff
@@ -425,9 +425,9 @@ def parse_structure(key, group, value, _type, compression):
             try:
                 group.attrs[key] = value
             except:
-                print("The hdf5 writer could not write the following "
-                      "information in the file")
-                print('%s : %s' % (key, value))
+                warnings.warn(
```

Comment:
...and I would probably use `logging.error` 
 >"Due to a more serious problem, the software has not been able to perform some function." 

seems to be quite accurate